lib/surface/compiler.ex

```diff
@@ -718,33 +716,80 @@ defmodule Surface.Compiler do
     :ok
   end
 
-  defp missing_slot(mod, slot_name, template_meta, parent_meta) do
+  defp missing_slot(module, slot_name, meta, _defined_slot_names, true) do
+    message = """
+    no slot `#{slot_name}` defined in the component `#{module}`\
+    \n\nPlease declare the default slot using `slot default` in order to use the `<slot />` notation.
+    """
+
+    IOHelper.compile_error(message, meta.file, meta.line)
+  end
+
+  defp missing_slot(module, slot_name, meta, defined_slot_names, false) do
+    similar_slot_message = similar_slot_message(slot_name, defined_slot_names)
+
+    existing_slots_message = existing_slots_message(defined_slot_names)
+
+    message = """
+    no slot `#{slot_name}` defined in the component `#{module}`\
+    #{similar_slot_message}\
+    #{existing_slots_message}\
+    \n\nHint: You can define slots using the `slot` macro. \
+    \nFor instance: `slot #{slot_name}`\
+    """
```

Comment:
I think it's easier to read without those `\n`s :)

```suggestion
    message = """
    no slot `#{slot_name}` defined in the component `#{inspect(module)}`\
    #{similar_slot_message}\
    #{existing_slots_message}
    
    Hint: You can define slots using the `slot` macro.
    
    For instance: `slot #{slot_name}`\
    """
```